### PR TITLE
feat: core snapshot vs db experimental

### LINKED ISSUE #236: Diff tool to compare core snapshot with datanode API
A new tool is required for comparing a core snapshot with datanode API. The tool takes the following parameters:

1. snapshot DB path - a path to the core snapshot database
2. datanode connection string - a connection string to a running data node

e.g. `vegatools difftool -s ./snapshot-tmp -d 127.0.0.1:3027`

The above assumes that no blocks are being produced after the last snapshot in the database such that we can expect that the snapshot state represents the current state that should be returned by datanode. 

The tool currently compares the following datasets:
1. account balances
2. diffOrders,
3. markets
4. parties
5. network limits
6. assets
7. delegations
8. epoch
9. vegatime
10. validator nodes data
11. network parameters
12. governance proposals
13. active deposits
14. active withdrawals
15. liquidity provisions
16. stake linking
17. banking transfers

TODO: positions

Need to consider porting to vega repo as `vega tool`.

### LINKED ISSUE #6184: Sanity check data in datanode vs snapshot core
Requirements
* Be able to take a snapshot database, and a datanode database and run a set of sanity data checks
* Return success or all of the mismatches 
* Initial set of comparisons: proposals, markets, transfers, account balances, delegations, orders, assets. 

Limitations
* Only a subset of the data in the snapshot is comparable - many things are transient in memory and will never be in the snapshot but will be expected in datanode, e.g. trades, rewards, etc. 
* This comparison can only work if the database is frozen, does not change, and includes the snapshot block updates as the last.

### PR DESCRIPTION
**Pull request by ze97286**
Closes #236

### CODE REVIEW

File: difftool/diff/diff_tool.go [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+		keys = append(keys, a.Id)
+		coreStr = append(coreStr, a.String())
+	}
+
+	return diff("proposal", coreStr, dbIdToTransfer, keys)
+}
+
+func diff(name string, core []string, db map[string]string, key []string) error {
+	errs := []string{}
+	for i, coreStr := range core {
```
Comment (wwestgarth) [inline difftool/diff/diff_tool.go]: Aren't we missing the case where something is in the data-node, but missing from the core-snapshot? i.e the core snapshot could be completely empty and the tool will say its fine, there is no diff.

Comment (ze97286) [reply difftool/diff/diff_tool.go]: 1. don't waste time looking at *this* version. let me push the new one. 
2. the purpose here to check the sanity of datanode treating the snapshot as the ground truth

Comment (ze97286) [reply difftool/diff/diff_tool.go]: I just pushed the latest version

File: difftool/diff/diff_test.go [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+	"testing"
+
+	v1 "code.vegaprotocol.io/vega/protos/vega/events/v1"
+)
+
+func TestXxx(t *testing.T) {
+	tt := []*v1.Transfer{}
+	tt = append(tt, &v1.Transfer{Id: "12345"}, &v1.Transfer{Id: "67890"})
+	println(sliceToString(tt))
+}
```
Comment (peterbarrow) [inline difftool/diff/diff_test.go]: Does this still need to be in here? It looks like local testing code.

**Review by peterbarrow — DISMISSED** [review]

File: difftool/diff/diff_test.go (hunk 2 above)
Comment (ze97286) [reply difftool/diff/diff_test.go]: but what a nice test it is!

Comment (ze97286) [reply difftool/diff/diff_test.go]: so elegant!

Comment (ze97286) [reply difftool/diff/diff_test.go]: removed

**Review by peterbarrow — DISMISSED** [review]

**Review by peterbarrow — APPROVED** [review]

**Pull request merged**